sdks/java/core/src/main/java/org/apache/beam/sdk/io/common/retry/RetryConfiguration.java

```diff
@@ -0,0 +1,80 @@
+package org.apache.beam.sdk.io.common.retry;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.io.Serializable;
+import org.joda.time.Duration;
+
+/**
+ * A POJO encapsulating a configuration for retry behavior when issuing requests. A retry will be
+ * attempted until the maxAttempts or maxDuration is exceeded, whichever comes first, for any of the
+ * following exceptions:
+ */
+public class RetryConfiguration implements Serializable {
+
+  private int maxAttempts;
+  private Duration maxDuration;
+  private RetryPredicate retryPredicate;
```

Comment:
The original design of `SolrIO` deliberately kept the predicate package private and visible only for tests.
That design was chosen to abide by the policy of Beam IOs which is to only expose tuning for those things that a user might reasonably need to tune. Personally I am not as strict as others, but I do think that in the case of retries the IO should be able to make the decision and this shouldn't be exposed to the user to modify.

Reply:
+1 to that, indeed you described well the no knobs philosophy of Beam